### ISSUE TITLE
Fix multiple declarations

### ISSUE THREAD

**Issue by jjhursey** [pmix/pmix-standard#264]
PR #263 introduced a new 'make check-decl' that turned up a few multiple declaration items that need to be cleaned up.

```
shell$ make check-decl
====> Checking for Multi-declared items
Error: PMIX_ERR_JOB_NO_EXE_SPECIFIED declared 2 times
Error: PMIX_ERR_JOB_SENSOR_BOUND_EXCEEDED declared 2 times
Error: PMIX_JOB_STATE_LAUNCH_UNDERWAY declared 2 times
Error: PMIX_ERR_JOB_ALLOC_FAILED declared 2 times
Error: PMIX_ERR_JOB_FAILED_TO_LAUNCH declared 2 times
Error: PMIX_ERR_JOB_ABORTED declared 2 times
Error: PMIX_JOB_STATE_UNDEF declared 2 times
Error: PMIX_JOB_STATE_SUSPENDED declared 2 times
Error: PMIX_GROUP_DECLINE declared 2 times
Error: PMIX_ERR_JOB_NON_ZERO_TERM declared 2 times
Error: PMIX_JOB_STATE_TERMINATED declared 2 times
Error: PMIX_ERR_JOB_ABORTED_BY_SIG declared 2 times
Error: PMIX_ERR_JOB_APP_NOT_EXECUTABLE declared 2 times
Error: PMIX_ERR_JOB_KILLED_BY_CMD declared 2 times
Error: PMIX_GROUP_ACCEPT declared 2 times
Error: PMIX_JOB_STATE_CONNECTED declared 2 times
Error: PMIX_ERR_INVALID_OPERATION declared 2 times
Error: PMIX_ERR_JOB_TERM_WO_SYNC declared 2 times
Error: PMIX_ERR_JOB_FAILED_TO_MAP declared 2 times
Error: PMIX_JOB_STATE_RUNNING declared 2 times
Error: PMIX_JOB_STATE_PREPPED declared 2 times
Error: PMIX_JOB_STATE_TERMINATED_WITH_ERROR declared 2 times
Error: pmix_job_state_t declared 2 times
Error: pmix_data_array_t declared 2 times
Error: PMIX_DATA_ARRAY_CONSTRUCT declared 2 times
Error: PMIX_DATA_ARRAY_CREATE declared 2 times
Error: PMIX_DATA_ARRAY_DESTRUCT declared 2 times
Error: PMIX_DATA_ARRAY_FREE declared 2 times
Error: PMIx_Job_state_string declared 2 times
--------------------------------------------------
Found 29 number of multiple declares
make: *** [check-decl] Error 29
```

**Comment by jjhursey** [comment]
Looking at v4:

```
make check-decl
====> Checking for Multi-declared items
Error: PMIX_ERROR_GROUP_LOCAL declared 2 times
Error: PMIX_ERROR_HANDLER_ID declared 2 times
Error: PMIX_ERROR_NAME declared 2 times
Error: PMIX_LOCALITY declared 2 times (Needs fixed)
Error: PMIX_ERROR_GROUP_SPAWN declared 2 times
Error: PMIX_ERROR_GROUP_MIGRATE declared 2 times
Error: PMIX_ERROR_GROUP_NODE declared 2 times
Error: PMIX_ERROR_GROUP_COMM declared 2 times
Error: PMIX_ERROR_GROUP_GENERAL declared 2 times
Error: PMIX_ERROR_GROUP_RESOURCE declared 2 times
Error: PMIX_SERVER_SCHEDULER declared 2 times (needs fixed)
Error: PMIX_ERROR_GROUP_ABORT declared 2 times
--------------------------------------------------
Found 12 number of multiple declares
make: *** [check-decl] Error 12
```

With the exception of the two below all of the above are deprecated+removed references. I fixed the two below in PR #304
 * `PMIX_LOCALITY`
 * `PMIX_SERVER_SCHEDULER`

**Comment by jjhursey** [comment]
PR #304 resolved this issue. This check is now part of the normal 'make' process so CI should catch any regressions.
